File: Server/src/main/content/global/skill/production/cooking/recipe/topping/impl/UncookedEgg.java
package content.global.skill.production.cooking.recipe.topping.impl;

import content.global.skill.production.cooking.recipe.topping.ToppingRecipe;
import core.game.node.item.Item;

/**
 * The Uncooked egg.
 */
public final class UncookedEgg extends ToppingRecipe {

    private static final Item EGG = new Item(1944);

    private static final Item UNCOOKED_EGG = new Item(7076);

    @Override
    public int getLevel() {
        return 1;
    }

    @Override
    public double getExperience() {
        return 1;
    }

    @Override
    public Item getProduct() {
        return UNCOOKED_EGG;
    }

    @Override
    public Item[] getIngredients() {
        return new Item[]{EGG};
    }

    @Override
    public boolean isSingular() {
        return true;
    }

    @Override
    public Item[] getParts() {
        return new Item[]{};
    }

}

File: Server/src/main/content/global/skill/production/cooking/recipe/topping/impl/UncookedEgg.kt
package content.global.skill.production.cooking.recipe.topping.impl

import content.global.skill.production.cooking.recipe.topping.ToppingRecipe
import cfg.consts.Items
import core.game.node.item.Item

/**
 * Uncooked egg.
 */
class UncookedEgg : ToppingRecipe() {

    override fun getLevel(): Int {
        return 1
    }

    override fun getExperience(): Double {
        return 1.0
    }

    override fun getProduct(): Item {
        return UNCOOKED_EGG
    }

    override fun getIngredients(): Array<Item> {
        return arrayOf(EGG)
    }

    override fun isSingular(): Boolean {
        return true
    }

    override fun getParts(): Array<Item> {
        return emptyArray()
    }

    companion object {
        private val EGG = Item(Items.EGG_1944)
        private val UNCOOKED_EGG = Item(Items.UNCOOKED_EGG_7076)
    }
}

